code/network/multimsgs.cpp

```diff
@@ -9114,3 +9115,133 @@ void process_sexp_packet(ubyte *data, header *hinfo)
 
 	sexp_packet_received(received_packet, num_ubytes);
 }
+
+// TODO, calculate how many we can send.
+constexpr ubyte MAX_TURRET_TRACK_CHANGES = 32;
+constexpr int SIZE_LOCATION_IN_PACKET = 7;
+
+SCP_vector<int> send_turret_tracking_packet(ushort parent_sig, SCP_vector<int>& subsys_indexes) 
+{
+	Assertion(MULTIPLAYER_MASTER, "A non-server is trying to send a turret tracking packet. This is a coder mistake, please report!");
+
+	SCP_vector<int> overflow_list;
+
+	// somehow trying to send for an an invalid or non-network object
+	if (parent_sig == 0) {
+		// returning an empty list will signal the manager to clear its entries
+		return overflow_list;
+	}
+
+	auto temp = multi_get_network_object(parent_sig);
+
+	// this means that the ship may have been destroyed before being able to send the packet.  Ignore.
+	if (temp == nullptr){
+		// returning an empty list will signal the manager to clear its entries
+		return overflow_list;
+	}
+
+	ship* shipp = &Ships[temp->instance];
+
+	// Required for all send functions
+	ubyte data[MAX_PACKET_SIZE];
+	int packet_size = 0;
+
+	BUILD_HEADER(TURRET_TRACK);
+
+	// send the time that this happened.
+	int time_out = Multi_Timing_Info.get_current_time();
+	ADD_INT(time_out);
+	ADD_USHORT(parent_sig);
+
+	ubyte last_index = (subsys_indexes.size() < MAX_TURRET_TRACK_CHANGES) ? static_cast<ubyte>(subsys_indexes.size()) : MAX_TURRET_TRACK_CHANGES;
+
+	// this is not usually done, but if the number of subsystems actually ends up being different than limit,
+	// this ubyte in the packet will get overwritten at the end of this function.
+	ADD_DATA(last_index);
```

Comment:
The subsys count should be the first thing in the packet to make it easier to update. Move it first, get rid of the magic number for its position, and just use `HEADER_LENGTH` as the offset when updating.

And to avoid confusion it should really be set to zero by default instead of the max possible count;